src/meta/meta_ckpt.c

```diff
@@ -236,37 +228,28 @@ __wt_meta_block_metadata(WT_SESSION_IMPL *session, const char *config, WT_CKPT *
     WT_DECL_RET;
     WT_KEYED_ENCRYPTOR *kencryptor;
     size_t encrypt_size, metadata_len;
-    char *min_config;
     const char *metadata, *filecfg[] = {WT_CONFIG_BASE(session, file_meta), NULL, NULL};
 
-    min_config = NULL;
     WT_ERR(__wt_scr_alloc(session, 0, &a));
     WT_ERR(__wt_scr_alloc(session, 0, &b));
 
+    /* Fill out the configuration array for normal retrieval. */
+    filecfg[1] = config;
+
     /*
+     * Find out if this file is encrypted. If encrypting, encrypt and encode.
      * The metadata has to be encrypted because it contains private data
      * (for example, column names). We pass the block manager text that
      * describes the metadata (the encryption information), and the
      * possibly encrypted metadata encoded as a hexadecimal string.
      * configuration string.
-     *
-     * Get a minimal configuration string, just the non-default entries.
-     */
-    WT_ERR(__wt_config_discard_defaults(session, filecfg, config, &min_config));
-
-    /* Fill out the configuration array for normal retrieval. */
-    filecfg[1] = config;
-
-    /*
-     * Find out if this file is encrypted. If encrypting, encrypt and encode the minimal
-     * configuration.
      */
     WT_ERR(__wt_btree_config_encryptor(session, filecfg, &kencryptor));
     if (kencryptor == NULL) {
-        metadata = min_config;
-        metadata_len = strlen(min_config);
+        metadata = config;
+        metadata_len = strlen(config);
     } else {
-        WT_ERR(__wt_buf_set(session, a, min_config, strlen(min_config)));
+        WT_ERR(__wt_buf_set(session, a, config, strlen(config)));
```

Comment:
Given the cost of encryption, it seems like we only we should only encrypt once -- perhaps `__conn_dhandle_config_set()` should call one routine that both fills in the base configuration value for the handle, and additionally encrypts a copy for later use here, and both strings are maintained in the dhandle for the life of the handle.

That also gets us away from using the current metadata configuration for anything other than retrieving checkpoint information -- in other words, if we believe the base configuration can't change, let's double-down on that belief.

Reply:
In `ckpt_set` we (after debugging) retain the old path and only use the faster path for checkpoints. There are a number of times and places where we cannot use that base and need that.

But WRT encryption, my changes do not change anything regarding encryption, only the string we're encrypting. However, I'm not following your suggestion. The base string stored in the dhandle alone (i.e. without the checkpoint and checkpoint_lsn elements) would encrypt differently than the resulting string with that base and those items concatenated to its end. It's not like we can just encrypt the `checkpoint=...,checkpoint_lsn` piece separately and concatenate that to the encrypted base-only string. We need to encrypt the whole string at once.